prov/sockets/src/sock_ep_msg.c

```diff
@@ -693,8 +694,10 @@ static void *sock_msg_ep_listener_thread(void *data)
 	poll_fds[0].events = poll_fds[1].events = POLLIN;
 
 	while (*((volatile int*) &ep->attr->cm.do_listen)) {
+		fastlock_acquire(&ep->attr->cm.lock);
 		timeout = dlist_empty(&ep->attr->cm.msg_list) ? -1 :
 					SOCK_CM_COMM_TIMEOUT;
+		fastlock_release(&ep->attr->cm.lock);
```

Comment:
I wonder if the correct solution here is to re-write the "you should keep listening" mechanism.

Indeed, this reading/writing to `do_listen` causes a helgrind warning in itself which I didn't report here.

Perhaps when it's time to stop listening, a control message should be sent on the listening fd of `hdr.type` `SOCK_TIME_TO_STOP_LISTENING`, which would trigger leaving the loop and ultimately quitting the listening thread.

In this way, the timeout can *always* be -1 for the `poll()` -- it doesn't matter if there's anything on the list or not.

Thoughts?